rasa/core/processor.py

```diff
@@ -540,19 +561,14 @@ def should_predict_another_action(action_name: Text) -> bool:
     async def _cancel_reminders(
         events: List[Event], tracker: DialogueStateTracker
     ) -> None:
-        """Cancel reminders by action_name"""
+        """Cancel reminders that match the ReminderCancelled event"""
 
         # All Reminders with the same action name will be cancelled
         for e in events:
             if isinstance(e, ReminderCancelled):
-                name_to_check = (
-                    str(e.action_name)
-                    + ACTION_NAME_SENDER_ID_CONNECTOR_STR
-                    + tracker.sender_id
-                )
                 scheduler = await jobs.scheduler()
                 for j in scheduler.get_jobs():
-                    if j.name == name_to_check:
+                    if e.matches_job_string(j.name, tracker.sender_id):
```

Comment:
I'd describe less *what* you're doing but more, why (intention)
```suggestion
                    if event.is_canceling_job(scheduled_job.name, tracker.sender_id):
```